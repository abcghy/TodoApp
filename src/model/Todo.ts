interface Todo {
    id: string;
    value: string;
    isDone: boolean;
}

export default Todo;