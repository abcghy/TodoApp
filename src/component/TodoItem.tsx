import Todo from "../model/Todo";
import Checkbox from "./Checkbox";

interface TodoItemProps {
  todo: Todo;
  onDone: (todo: Todo) => void;
  onDelete: (todo: Todo) => void;
}

function TodoItem({ todo, onDone, onDelete }: TodoItemProps) {
  return (
    <div
      className="flex items-center align-center cursor-pointer py-1 px-1 rounded group hover:(bg-red-50) active:(transform translate-y-px bg-red-100)"
      onClick={(e) => {
        onDone(todo);
        e.preventDefault();
      }}
    >
      <Checkbox isChecked={todo.isDone} setChecked={() => {}} />
      <div className={`ml-1 flex-1 text-xl ${todo.isDone && "line-through"}`}>
        {todo.value}
      </div>
      <div
        className="invisible group-hover:visible text-white bg-hex-DB4D37 rounded px-1 py-0.5 select-none text-sm hover:(bg-hex-B92B15) active:(bg-hex-970903 transform translate-y-px)"
        onClick={(e) => {
          e.stopPropagation();
          e.preventDefault();
          onDelete(todo);
        }}
      >
        删除
      </div>
    </div>
  );
}

export default TodoItem;
