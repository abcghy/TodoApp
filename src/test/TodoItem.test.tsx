import TodoItem from "../component/TodoItem";
import renderer from "react-test-renderer";
import Todo from "../model/Todo";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

it("test todo item display and listener", () => {
  let todo: Todo = {
    id: "1",
    value: "Read a book every month",
    isDone: false,
  };

  function onDone(todo: Todo) {
    console.log("onDone");
  }

  function onDelete(todo: Todo) {
    console.log("onDelete");
  }

  let component = renderer.create(
    <TodoItem todo={todo} onDone={onDone} onDelete={onDelete} />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  //#region isDone is true
  todo.isDone = true;
  component = renderer.create(
    <TodoItem todo={todo} onDone={onDone} onDelete={onDelete} />
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  //#endregion
});

it("check text and event", () => {
  let todo: Todo = {
    id: "1",
    value: "Read a book every month",
    isDone: false,
  };
  function onDone(todo: Todo) {
    console.log("onDone");
  }
  function onDelete(todo: Todo) {
    console.log("onDelete");
  }
  render(<TodoItem todo={todo} onDone={onDone} onDelete={onDelete} />);
  expect(screen.getByText("Read a book every month")).toBeInTheDocument();
});
