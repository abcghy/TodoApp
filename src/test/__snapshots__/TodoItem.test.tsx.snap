// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test todo item display and listener 1`] = `
<div
  aria-describedby=""
  aria-roledescription="sortable"
  className="flex items-center align-center cursor-pointer py-1 px-1 relative rounded group hover:(bg-red-50) active:(transform translate-y-px bg-red-100)"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "transform": undefined,
      "transition": undefined,
      "zIndex": 0,
    }
  }
  tabIndex={0}
>
  <div
    className="cursor-pointer my-auto"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc>
        Download more icon variants from https://tabler-icons.io/i/square
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <rect
        height="16"
        rx="2"
        width="16"
        x="4"
        y="4"
      />
    </svg>
  </div>
  <div
    className="ml-1 flex-1 text-xl false"
  >
    Read a book every month
  </div>
  <div
    className="invisible group-hover:visible text-white bg-hex-DB4D37 rounded px-1 py-0.5 select-none text-sm hover:(bg-hex-B92B15) active:(bg-hex-970903 transform translate-y-px)"
    onClick={[Function]}
  >
    删除
  </div>
</div>
`;

exports[`test todo item display and listener 2`] = `
<div
  aria-describedby=""
  aria-roledescription="sortable"
  className="flex items-center align-center cursor-pointer py-1 px-1 relative rounded group hover:(bg-red-50) active:(transform translate-y-px bg-red-100)"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "transform": undefined,
      "transition": undefined,
      "zIndex": 0,
    }
  }
  tabIndex={0}
>
  <div
    className="cursor-pointer my-auto"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc>
        Download more icon variants from https://tabler-icons.io/i/square-check
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <rect
        height="16"
        rx="2"
        width="16"
        x="4"
        y="4"
      />
      <path
        d="M9 12l2 2l4 -4"
      />
    </svg>
  </div>
  <div
    className="ml-1 flex-1 text-xl line-through"
  >
    Read a book every month
  </div>
  <div
    className="invisible group-hover:visible text-white bg-hex-DB4D37 rounded px-1 py-0.5 select-none text-sm hover:(bg-hex-B92B15) active:(bg-hex-970903 transform translate-y-px)"
    onClick={[Function]}
  >
    删除
  </div>
</div>
`;
